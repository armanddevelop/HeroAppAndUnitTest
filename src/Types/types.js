export const types = {
  login: "[aut] Login",
  logout: "[aut] Logout",
};
