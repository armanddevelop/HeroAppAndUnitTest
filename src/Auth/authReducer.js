import { types } from "../Types/types";

// const initialState = {
//   name:"Licha",
//   logged: true
// }

export const authReducer = (state = {}, action) => {
  console.log("Esto vale action ", action);
  switch (action.type) {
    case types.login:
      return {
        ...action.payload,
        logged: true,
      };
    case types.logout:
      return {
        logged: false,
      };
    default:
      return state;
  }
};
