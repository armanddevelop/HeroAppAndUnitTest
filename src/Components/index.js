export { HeroList } from "./Hero/HeroList";

export { Hero } from "./Hero/Hero";

export { Navbar } from "./UI/NavBar";

export { Login } from "./Login/Login";

export { SearchScreen } from "./Search/SearchScreen";

export { DcScreen } from "./DC/DcScreen";

export { MarvelScreen } from "./Marvel/MarvelScreen";
