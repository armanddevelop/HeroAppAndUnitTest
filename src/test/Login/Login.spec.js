import { mount } from "enzyme";
import { MemoryRouter } from "react-router-dom";
import { AuthContext } from "../../Auth/AuthContext";
import { Login } from "../../Components";
import { types } from "../../Types/types";

const mockNavigate = jest.fn();
const dispatch = jest.fn();
jest.mock("react-router-dom", () => ({
  ...jest.requireActual("react-router-dom"),
  useNavigate: () => mockNavigate,
}));
describe("Test in Login component", () => {
  const contextValue = {
    dispatch,
  };
  const container = mount(
    <AuthContext.Provider value={contextValue}>
      <MemoryRouter initialEntries={["/login"]}>
        <Login />
      </MemoryRouter>
    </AuthContext.Provider>
  );
  it("Snapshot to Login component", () => {
    expect(container).toMatchSnapshot();
  });
  it("Should run handleLogIn without path in localStorage", () => {
    const action = {
      type: types.login,
      payload: { name: "alicha" },
    };
    container.find("button").prop("onClick")();

    expect(dispatch).toHaveBeenCalledWith(action);
    expect(mockNavigate).toHaveBeenCalledWith("", { replace: true });
  });
  it("Should run handleLogIn with path in localStorage", () => {
    const pathObj = {
      pathname: "/dc",
      search: "",
      hash: "",
      state: null,
      key: "default",
    };
    const myObjString = JSON.stringify(pathObj);
    localStorage.setItem("lastPath", myObjString);
    container.find("button").prop("onClick")();

    expect(mockNavigate).toHaveBeenCalledWith(pathObj.pathname, {
      replace: true,
    });
  });
});
