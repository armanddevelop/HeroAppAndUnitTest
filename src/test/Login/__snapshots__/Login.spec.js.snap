// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in Login component Snapshot to Login component 1`] = `
<div
  className="container mt-5"
>
  <h1>
    Login
  </h1>
  <hr />
  <button
    className="btn btn-primary"
    onClick={[Function]}
  >
    LogIn
  </button>
</div>
`;
